gradle-palantir-java-format/src/main/groovy/com/palantir/javaformat/gradle/PalantirJavaFormatPlugin.java

```diff
@@ -16,42 +16,77 @@
 
 package com.palantir.javaformat.gradle;
 
+import com.palantir.javaformat.bootstrap.NativeImageFormatterService;
 import com.palantir.javaformat.java.FormatterService;
 import java.io.IOException;
+import java.nio.file.Path;
 import org.gradle.api.DefaultTask;
 import org.gradle.api.Plugin;
 import org.gradle.api.Project;
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
+import org.gradle.api.provider.Property;
+import org.gradle.api.provider.Provider;
+import org.gradle.api.tasks.Input;
 import org.gradle.api.tasks.TaskAction;
 
 public final class PalantirJavaFormatPlugin implements Plugin<Project> {
 
     @Override
     public void apply(Project project) {
         project.getRootProject().getPlugins().apply(PalantirJavaFormatProviderPlugin.class);
+        project.getRootProject().getPlugins().apply(NativeImageFormatProviderPlugin.class);
         project.getRootProject().getPlugins().apply(PalantirJavaFormatIdeaPlugin.class);
 
         project.getPlugins().apply(PalantirJavaFormatSpotlessPlugin.class);
 
         project.getPlugins().withId("java", p -> {
-            project.getTasks().register("formatDiff", FormatDiffTask.class);
 
             // TODO(dfox): in the future we may want to offer a simple 'format' task so people don't need to use
             // spotless to try out our formatter
+            project.getTasks().register("formatDiff", FormatDiffTask.class, task -> {
+                if (NativeImageFormatProviderPlugin.isNativeImageSupported()
+                        && NativeImageFormatProviderPlugin.shouldUseNativeImage(project)) {
```

Comment:
Perhaps we should just make `shouldUseNativeImage` include `isNativeImageSupported`. I don't think we're ever going to use the native image if it's not supported?